lib/client.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const nodeVault = require('node-vault')
+
+class HashiCorpClient {
+  /**
+   * @param options {{mountPoint?: string, vaultOptions: import("node-vault").VaultOptions}}
+   */
+  constructor({ mountPoint = 'secret', vaultOptions } = {}) {
+    this.mountPoint = mountPoint
```

Comment:
mountPoint is optional, right? if the user provides only vaultOptions then mountPoint will be undefined, and the template string you're using to access the secret will be `"undefined/${name}"`

Reply:
mountPoint should default to secret if they don't provide it:

<img width="481" alt="Screenshot 2021-02-19 at 11 49 45" src="https://user-images.githubusercontent.com/1027024/108501020-8ff1e900-72a8-11eb-9f81-6edc2d12535e.png">

Reply:
yep my bad, misread the code. as suggested by matteo though, to avoid confusion, move it down and make the default explicit